analysis_options.yaml

```diff
@@ -42,7 +42,7 @@ linter:
     # - avoid_bool_literals_in_conditional_expressions # not yet tested
     # - avoid_catches_without_on_clauses # we do this commonly
     # - avoid_catching_errors # we do this commonly
-    - avoid_classes_with_only_static_members
+    # - avoid_classes_with_only_static_members # required for PluginUtilities caching / lookup methods
```

Comment:
Please don't comment out this line, though feel free to add additional comments based on the following to explain why this is set:

We want to avoid classes that can be instantiated but only have statics, as in:

```dart
class A { static void test() { } }
```

The way we avoid allowing this to be instantiated is to have an invalid private factory constructor. The pattern is:

```dart
class A {
  // This class is only a namespace, and should not be instantiated or
  // extended directly.                                                                    
  factory A._() => null;                                            
  void test() { }
}
```